src/components/PeoplePage.tsx

```diff
@@ -1,34 +1,122 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [peopleLoader, setPeopleLoader] = useState(true);
+  const [error, setError] = useState(false);
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const loadPeople = async () => {
+    try {
+      setError(false);
+      setPeopleLoader(true);
+      const load = await getPeople();
+
+      setPeople(load);
+    } catch {
+      setError(true);
+    }
+
+    setPeopleLoader(false);
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const noPeopleOnServer = () => {
+    return !people.length && !error && !peopleLoader;
+  };
+
+  const peopleFiltered = () => {
+    let filtered = [...people];
+
+    if (centuries.length) {
+      filtered = [
+        ...filtered.filter(
+          person => centuries.some(
+            century => {
+              return Math.ceil(person.born / 100)
+              === Number(century);
+            },
+          ),
+        ),
+      ];
+    }
+
+    if (query) {
+      filtered = [...filtered.filter(({
+        name,
+        fatherName,
+        motherName,
+      }) => (
+        name.toLowerCase().includes(query.toLowerCase())
+        || fatherName?.toLowerCase().includes(query.toLowerCase())
+        || motherName?.toLowerCase().includes(query.toLowerCase())
```

Comment:
move this check to separate helper to avoid code duplicates, like this:

```typescript
const isMatch = (str: string) => str.toLowerCase().includes(query.toLowerCase())
```